### DISCUSSION TITLE
Restart daily

### DISCUSSION THREAD

**Discussion by HolgerM1** [vogler/free-games-claimer#56] [Q&A] [answered]
Hi,
Great Stuff! Can anybody give me a hint how to use crontab to set up daily restarts of the script? I don't have much experience with linux, sorry. I'm running docker on an Ubuntu VM.

Thanks!

**Comment by vogler** [comment, marked as answer]
Run `crontab -e` to edit your cronjobs and add

```
#  m h  dom mon dow command
  55 19 *   *   *   docker run --rm -p 6080:6080 -v fgc:/fgc/data --pull=always ghcr.io/vogler/free-games-claimer
```
which will run it every day at 19:55.

Note that the scheduled command does not include `-it` which is only needed for the initial login prompts.

**Comment by HolgerM1** [comment]
thank you very much!

**Comment by HolgerM1** [comment]
May I ask another question on that issue?!
I tried this:
![Screenshot 2023-02-11 084313](https://user-images.githubusercontent.com/114694650/218250505-e386fd87-d2a9-4a08-9888-7e1c512265bc.png)

leading to this:
![Screenshot 2023-02-11 084148](https://user-images.githubusercontent.com/114694650/218250533-8817bd4a-7131-4df9-871b-9c572aa53df3.png)

What am I doing wrong?

**Reply by vogler** [reply]
Ah, try dropping `-it`. It's only needed for the login prompts.
https://docs.docker.com/engine/reference/commandline/run/

**Comment by HolgerM1** [comment]
Thanks again, I meanwhile found this:
[https://ismailyenigul.medium.com/docker-error-in-crontab-the-input-device-is-not-a-tty-7280cc42cf19](url)
and this
[https://www.dennisotugo.com/fix-the-input-device-is-not-a-tty-docker/](url)

I was not sure which solution was correct -i or --tty. Can I just leave it blank?

**Reply by vogler** [reply]
Yes, just remove `-it` since you don't need to interact with it while it's running.

**Comment by TRusselo** [comment]
editing CRON on unraid systems is not advised.
it does not survive reboot.

on Unraid, it is recommended to use "user scripts" plugin

**Reply by osnapitsjoey** [reply]
should I use the same command up above?

**Reply by rorar** [reply]
Using the Plugin "User Scripts", adding a daily schedule and the following command:
`docker start free-games-claimer`
should do the trick.

**Comment by HolgerM1** [comment]
It'r running on an Proxmox server?!

**Reply by vogler** [reply]
Idk, why shouldn't it?

**Comment by frdmn** [comment]
If you are working with a `compose.yml` file, you can the following crontab instead:

```
55 19 *   *   *   docker compose -f /opt/docker/epic-claimer/compose.yml up -d
```

**Comment by K2ouMais** [comment]
If you want to run it from a ```docker-compose.yml``` file, this is the best approach.

Open the user crontab with ```crontab -e``` and insert the following line at the very end of the crontab.

```
30 17 * * * docker compose -f ~/.projects/free-games-claimer/docker-compose.yml run --rm free-games-claimer
```

It will start every day the container at 17:30 and remove the running container afterwards.

Edit the path to your ```docker-compose.yml``` file and you are good to go.

**Comment by osnapitsjoey** [comment]
Well I already have the docker installed, so could I just point the path to
my already installed one? Because if I don't it seems to install another
instance

**Reply by vogler** [reply]
I don't get what you're asking. Do you want to use docker compose? If so, run it in the path where your `docker-compose.yml` is or supply its path like mentioned with `-f ...`. The image will only be downloaded once, no matter for how many containers you use it.

If you have no specific reason for using docker compose, you can just use `docker run` like mentioned in my marked response.

**Reply by osnapitsjoey** [reply]
sorry for the late reply. no I want to run this via userscripts. im assuming i just place "  55 19 *   *   *   docker run --rm -p 6080:6080 -v path/to/fgc
"
and call it a day?

**Comment by Stooovie** [comment]
I do have FGC working, I do have properly set cron:
`  55 19 *   *   *   docker run --rm -p 6080:6080 -v fgc:/fgc/data --pull=always ghcr.io/vogler/free-games-claimer bash -c "node epic-games.js; node gog.js"`
yet the script doesn't run every day for some reason and I don't know how to investigate further. Every time I run it manually, it requires logging again into Epic. Running in a proxmox lxc.

**Reply by vogler** [reply]
Did you use `crontab -e`? It only updates on exit. Try writing a line to a log file via a cronjob, then try it with your `docker run` command. Check the log if it doesn't work.

> Every time I run it manually, it requires logging again into Epic.

Are you sure that nothing is deleting your `fgc` volume? Maybe running `docker prune` or some other cleanup in Docker Desktop?

**Comment by mgrimace** [comment]
If you're using compose, and you want the daily scheduling to be self-contained (i.e., no cron), this compose appears to work for me. I'm currently testing, but I set it yesterday and it claimed a game for me today:

```yaml

services:
  free-games-claimer:
    container_name: fgc
    image: ghcr.io/vogler/free-games-claimer
    build: .  
    ports:
      - 6080:6080
    environment:
      - TZ=America/Toronto
    env_file:
      - /opt/appdata/fgc/.secrets.env
    volumes:
      - /opt/appdata/fgc:/fgc/data #your config file location
    command: bash -c "node prime-gaming; node gog; echo sleeping; sleep 1d; docker restart fgc" #add node epic; if you want to try epic again
    restart: unless-stopped
```
